### PR TITLE
Vehicle control

### PR DESCRIPTION
**Pull request by bminortx**
Lots of changes to the readme. Also cleans up the urdf file system.

### CODE REVIEW

File: Applications/LocalSim/LocalSim.cpp [hunk 1]
```diff
@@ -42,7 +37,7 @@ LocalSim::LocalSim(const string& sLocalSimName,
   // 4. We must decide the next actions based off of the Server Option.
   cout<<" The server option is set to "<<sServerOption<<"."<<endl;
 
-  m_NetworkManager.Init( m_sLocalSimName, sServerOption, 0);
+  m_NetworkManager.Init( m_sLocalSimName, sServerOption, 1);
```
Comment (chachi) [inline Applications/LocalSim/LocalSim.cpp]: Remove the space between `(` and `m_sLocalSimName`?

File: Applications/SimPlanner/PlannerLibTest.h [hunk 2]
```diff
@@ -38,6 +38,8 @@ class KeyCarController{
       phi = phi_.at(count_);
     }
     bool applied = car_->ApplyCommand(force, phi);
+    std::cout<<force<<std::endl;
```
Comment (chachi) [inline Applications/SimPlanner/PlannerLibTest.h]: Do you want to leave these cout's in?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+SimPlanner
+---------
+Work in progress...
+
+******************************
+
+MAIN DEVELOPERS:
+----------
+* Brandon Minor
```
Comment (chachi) [inline README.md:84]: That's a nice README. :+1:

File: SimBA/Managers/NetworkManager.cpp [hunk 4]
```diff
@@ -15,18 +14,17 @@ bool NetworkManager::Init(string sProxyName, string sServerName, int verbosity){
...
   }
-
   // If we do need it, Init Node connection in LocalSim and provide
   // relative RPC method.
   m_sLocalSimName  = sProxyName;
   m_iNodeClients   = 0;
-
-  m_Node.set_verbosity(m_verbosity); // make some noise on errors
-  bool worked = m_Node.init(m_sLocalSimName);
+  node_.set_verbosity(1); // make some noise on errors
```
Comment (chachi) [inline SimBA/Managers/NetworkManager.cpp]: Still doesn't make noise on errors :-)

File: SimBA/Managers/NetworkManager.cpp [hunk 5]
```diff
@@ -15,18 +14,17 @@ bool NetworkManager::Init(string sProxyName, string sServerName, int verbosity){
...
+  bool worked = node_.init(m_sLocalSimName);
   if(!worked && m_verbosity!=0){
     cout<<"[NetworkManager] Init Node '"<<m_sLocalSimName<<"' failed..."<<endl;
     return false;
   }
+  cout<<"SUCCESS: Initialized node for "<<m_sLocalSimName<<endl;
```
Comment (chachi) [inline SimBA/Managers/NetworkManager.cpp]: Do you have miniglog in here/do you want all these cout's printing?

File: SimBA/Managers/NetworkManager.cpp [hunk 6]
```diff
@@ -131,7 +128,6 @@ void NetworkManager::RegisterDevices(SimDevices* pSimDevices){
       Device->m_bDeviceOn = true;
     }
   }
-
   else{
```
Comment (chachi) [inline SimBA/Managers/NetworkManager.cpp]: Might want to move this up to be `} else {`

File: SimBA/Managers/NetworkManager.cpp [hunk 7]
```diff
@@ -240,26 +236,21 @@ void NetworkManager::RegisterSensorDevice(RegisterNodeCamReqMsg& mRequest,
...
+    cout<<"Attempting to subscribe to "<<
+        controller_name+"/"+controller_name<<endl;
+    int subscribe_try = 0;
+    while (node_.subscribe(controller_name+"/"+controller_name) != true &&
+           subscribe_try<500) {
+      cout<<".";
+      usleep(100);
+      subscribe_try++;
+    }
+    if(subscribe_try>=500){
```
Comment (chachi) [inline SimBA/Managers/NetworkManager.cpp]: Spacing around `>=`?

File: SimBA/Managers/NetworkManager.cpp [hunk 8]
```diff
@@ -566,7 +556,7 @@ bool NetworkManager::RegisterWithStateKeeper()
 {
   // 1. Subscribe to StateKeeper World state topic
   string sServiceName = m_sServerName+"/WorldState";
-  if( m_Node.subscribe(sServiceName) == false ){
+  if( node_.subscribe(sServiceName) == false ){
```
Comment (chachi) [inline SimBA/Managers/NetworkManager.cpp]: spacing around `(` and `)`?

File: SimBA/Managers/NetworkManager.cpp [hunk 9]
```diff
@@ -677,8 +667,8 @@ void NetworkManager::AddRobotByURDF(LocalSimAddNewRobotReqMsg& mRequest,
 
 void NetworkManager::DeleteRobot(LocalSimDeleteRobotReqMsg& mRequest,
                                  LocalSimDeleteRobotRepMsg& mReply){
-  // don't let anyone touch the shared resource table...
-  mutex lock();
+  // Don't let anyone touch the shared resource table...
+  mtx_.lock();
```
Comment (chachi) [inline SimBA/Managers/NetworkManager.cpp]: Use a `std::lock_guard<std::mutex>` to get automatic unlocking

File: SimBA/Managers/NetworkManager.cpp [hunk 10]
```diff
@@ -698,7 +690,7 @@ void NetworkManager::DeleteRobot(LocalSimDeleteRobotReqMsg& mRequest,
 /// pose, state, and current command.
 bool NetworkManager::PublishRobotToStateKeeper(){
   // don't let anyone touch the shared resource table...
-  mutex lock();
+  mtx_.lock();
```
Comment (chachi) [inline SimBA/Managers/NetworkManager.cpp]: Same comment with `lock_guard` here. Keeps you from remembering to call `unlock()` everywhere.

Comment (bminortx) [inline SimBA/Managers/NetworkManager.cpp]: Helpful; I didn't know that existed, haha.

File: SimBA/Managers/NetworkManager.cpp (hunk 5 above)
Comment (bminortx) [inline SimBA/Managers/NetworkManager.cpp]: I have miniglog now!

File: SimBA/Managers/NetworkManager.cpp (hunk 8 above)
Comment (bminortx) [inline SimBA/Managers/NetworkManager.cpp]: fixed in next push

**Pull request merged**